src/init.c

```diff
@@ -572,8 +614,10 @@ static void jl_resolve_sysimg_location(JL_IMAGE_SEARCH rel)
         jl_options.outputbc = abspath(jl_options.outputbc, 0);
     if (jl_options.machine_file)
         jl_options.machine_file = abspath(jl_options.machine_file, 0);
-    if (jl_options.project && strncmp(jl_options.project, "@.", strlen(jl_options.project)) != 0)
+    if (jl_options.project && (strcmp(jl_options.project, "@.") != 0 || strcmp(jl_options.project, "@") != 0))
```

Comment:
It's not quite useless, since the result is different from `strcmp` (strlen+1 would be closer, but only because it would then ignore the extra parameter), so this was just a misleading way to write it. I've translated it exactly here, but from looking at the PR (https://github.com/JuliaLang/julia/pull/28625#discussion_r212749047) I don't think this was actually the intent of the code.

Reply:
I see what's going on here; previously it was a shorthand to be able to match both `"@"` and `"@."`.  Carry on.